src/ssl/MemStats.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_SSL_MEMSTATS_H
+#define SQUID_SRC_SSL_MEMSTATS_H
+
+#if USE_OPENSSL
+
+#include "StatHist.h"
+#include "store/forward.h"
+
+namespace Ssl
+{
+
+/// Statistics for OpenSSL malloc-based memory management.
+
+/// Common statistics for a single memory management function
+/// (i.e., malloc(), realloc() or free())
+class MemStats
```

Comment:
This is a little different hierarchy (from what we discussed), but think it is better.